app/src/main/java/org/ieeeguc/ieeeguc/models/Committee.java

```diff
@@ -66,27 +66,51 @@ public static void getAllCommittees(final HTTPResponse HTTP_RESPONSE){
             public void onFailure(Call call, IOException e) {
 
                 // No Internet connection.
-                HTTP_RESPONSE.onFailure(-1,null);
+                MainActivity.UIHandler.post(new Runnable() {
+                    @Override
+                    public void run() {
+
+                        HTTP_RESPONSE.onFailure(-1,null);
+                    }
+                });
                 call.cancel();
             }
 
             @Override
             public void onResponse(Call call, Response response){
 
-                int statusCode = response.code();
+                final int statusCode = response.code();
 
                 try{
                     String responseBody = response.body().string();
-                    JSONObject body = new JSONObject(responseBody);
+                    final JSONObject body = new JSONObject(responseBody);
                     if(statusCode >= 200 && statusCode < 300){
-                        HTTP_RESPONSE.onSuccess(statusCode,body);
+                        MainActivity.UIHandler.post(new Runnable() {
```

Comment:
i think it will look neater if you moved the whole if condition inside the runnable